File: api/user-service/src/main/java/com/carrental/userservice/model/dto/UserUpdateDTO.java
package com.carrental.userservice.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.PastOrPresent;
import java.time.LocalDate;

@Data
public class UserUpdateDTO {

    @NotEmpty
    private String name;

    @NotEmpty
    private String surname;

    @NotEmpty
    private String phone;

    @NotNull
    @PastOrPresent
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate birthDate;
}

File: frontend-app/shared/model/UserUpdateDTO.ts
export class UserUpdateDTO {
    name: string;
    surname: string;
    phone: string;
    birthDate: string;

    constructor(name: string, surname: string, phone: string, birthDate: string) {
        this.name = name;
        this.surname = surname;
        this.phone = phone;
        this.birthDate = birthDate;
    }
}
